src/basic/fs-util.c

```diff
@@ -1043,6 +1043,29 @@ int open_mkdir_at(int dirfd, const char *path, int flags, mode_t mode) {
         return TAKE_FD(fd);
 }
 
+int open_mkdirp_at(int dirfd, const char *path, int flags, mode_t mode) {
```

Comment:
You can use `chase_and_openat()` with `CHASE_MKDIR_0755` to get the same effect I think.